cpp-package/example/mlp_csv_cpu.cpp

```diff
@@ -0,0 +1,188 @@
+    
+    // Start training
+    for (int iter = 0; iter < max_epoch; ++iter) {
+        int samples = 0;
+        train_iter.Reset();
+        
+        auto tic = std::chrono::system_clock::now();
+        while (train_iter.Next()) {
+            samples += batch_size;
+            auto data_batch = train_iter.GetDataBatch();
+            
+            /*
+             * The shape of data_batch.data is (batch_size, (num_mnist_features + 1))
+             * Need to reshape this data so that label column can be extracted from this data.
+             */
+            NDArray reshapedData = data_batch.data.Reshape(Shape((num_mnist_features + 1),batch_size));
+            
+            // Extract the label data by slicing the first column of the data and copy it to "label" arg.
+            reshapedData.Slice(0,1).Reshape(Shape(batch_size)).CopyTo(&args["label"]);
+            
+            // Extract the feature data by slicing the columns 1 to 785 of the data and copy it to "X" arg.
+            reshapedData.Slice(1,(num_mnist_features + 1)).Reshape(Shape(batch_size,num_mnist_features)).CopyTo(&args["X"]);
+            
+            // Compute gradients
+            exec->Forward(true);
+            exec->Backward();
+            // Update parameters
+            for (size_t i = 0; i < arg_names.size(); ++i) {
+                if (arg_names[i] == "X" || arg_names[i] == "label") continue;
+                opt->Update(i, exec->arg_arrays[i], exec->grad_arrays[i]);
+            }
+        }
+        auto toc = std::chrono::system_clock::now();
+        
+        Accuracy acc;
+        val_iter.Reset();
+        while (val_iter.Next()) {
+            auto data_batch = val_iter.GetDataBatch();
+            
+            /*
+             * The shape of data_batch.data is (batch_size, (num_mnist_features + 1))
+             * Need to reshape this data so that label column can be extracted from this data.
+             */
+            NDArray reshapedData = data_batch.data.Reshape(Shape((num_mnist_features + 1),batch_size));
+            
+            // Extract the label data by slicing the first column of the data and copy it to "label" arg.
+            NDArray labelData = reshapedData.Slice(0,1).Reshape(Shape(batch_size));
+            labelData.CopyTo(&args["label"]);
+            
+            // Extract the feature data by slicing the columns 1 to 785 of the data and copy it to "X" arg.
+            reshapedData.Slice(1,(num_mnist_features + 1)).Reshape(Shape(batch_size,num_mnist_features)).CopyTo(&args["X"]);
+            
+            // Forward pass is enough as no gradient is needed when evaluating
+            exec->Forward(false);
+            acc.Update(labelData, exec->outputs[0]);
+        }
+        float duration = std::chrono::duration_cast<std::chrono::milliseconds>
+        (toc - tic).count() / 1000.0;
+        LG << "Epoch: " << iter << " " << samples/duration << " samples/sec Accuracy: " << acc.Get();
```

Comment:
Can you follow the same format as the Python Examples. 
see this https://github.com/apache/incubator-mxnet/pull/12182#issuecomment-416634452